diff --git a/shareable_activity.py b/shareable_activity.py
index 46d02b2..1e8de97 100644
--- a/shareable_activity.py
+++ b/shareable_activity.py
@@ -28,8 +28,9 @@ class ShareableObject(dbus.service.Object):
     def SendMessageTo(self, busname, msg, kwargs):
         pass
 
+from sugarapp.widgets import SugarCompatibleActivity
 
-class ShareableActivity(activity.Activity):
+class ShareableActivity(SugarCompatibleActivity):
 
     '''
     A shareable activity.
@@ -47,7 +48,7 @@ class ShareableActivity(activity.Activity):
             service_path
         '''
 
-        activity.Activity.__init__(self, handle, *args, **kwargs)
+        SugarCompatibleActivity.__init__(self, handle, *args, **kwargs)
 
         self._sync_hid = None
         self._message_cbs = {}
